artifacts/definitions/Linux/Events/TrackProcesses.yaml

```diff
@@ -0,0 +1,201 @@
+       WHERE InstallAudit
+     LET _ <= SELECT *
+       FROM execve(
+         argv=['apt-get', '-y', '-o', 'Debug::pkgProblemResolver=yes', '--no-install-recommends', 'auditd'])
+       WHERE InstallAudit
+
+     LET AuditCtl(action, syscalls, key) = SELECT *
+       FROM foreach(
+         row=('b64', 'b32', ),
+         query={
+           SELECT *
+           FROM execve(
+             argv=['auditctl', action, 'exit,always', '-F', 'arch='+_value, '-S', syscalls, '-k', key])
+         })
+
+     LET _ <= SELECT *
+       FROM AuditCtl(action='-a',
+                     syscalls='execve',
+                     key=AuditKeyExecve)
+     LET _ <= SELECT *
+       FROM AuditCtl(action='-a',
+                     syscalls='exit,exit_group',
+                     key=AuditKeyExit)
+     LET _ <= atexit(
+         query={
+           SELECT *
+           FROM chain(
+             r1={
+               SELECT *
+               FROM AuditCtl(action='-d',
+                             syscalls='execve',
+                             key=AuditKeyExecve)
+             },
+             r2={
+               SELECT *
+               FROM AuditCtl(action='-d',
+                             syscalls='exit,exit_group',
+                             key=AuditKeyExit)
+             })
+           WHERE RemoveRules
+         })
+
+     LET Users <= memoize(
+         query={
+           SELECT 
+                  int(
+                    int=Uid) AS UID,
+                  User
+           FROM Artifact.Linux.Sys.Users()
+         },
+         key='UID')
+
+     LET UpdateQuery = SELECT *
+       FROM foreach(
+         row={
+           SELECT *
+           FROM audit()
+           WHERE AuditKeyExecve IN Tags OR AuditKeyExit IN Tags
+         },
+         query={
+           SELECT *
+           FROM switch(
+             start={
+               SELECT 
+                      Process.pid AS id,
+                      Process.ppid AS parent_id,
+                      'start' AS update_type,
+                      dict(
+                        Pid=Process.pid,
+                        Ppid=Process.ppid,
+                        Name=Process.name,
+                        StartTime=timestamp(
+                          string=Timestamp),
+                        EndTime=NULL,
+                        Username=get(
+                          item=Users,
+                          field=User.ids.uid).User,
+                        Exe=Process.exe,
+                        CommandLine=join(
+                          sep=' ',
+                          array=Process.args),
+                        CurrentDirectory=get(
+                          item=Process,
+                          member='CWD'),
+                        TerminalSessionId=Session) AS data,
+                      timestamp(
+                        string=Timestamp) AS start_time,
+                      NULL AS end_time
+               FROM scope()
```

Comment:
You probably dont need a SELECT here - just use a dict() directly?

Reply:
I agree, but I am not sure that the result will be any better or more readable?